src/dataflow/src/source/mod.rs

```diff
@@ -465,8 +465,8 @@ impl ConsistencyInfo {
                 &source_id.to_string(),
                 &worker_id.to_string(),
             ),
-            // we have never downgraded, so make sure the initial value is outside of our frequncy
-            time_since_downgrade: Instant::now() - timestamp_frequency,
+            // we have never downgraded, so make sure the initial value is outside of our frequency
+            time_since_downgrade: Instant::now() - timestamp_frequency - Duration::from_secs(1),
```

Comment:
I don't know how I never noticed earlier but I don't think `Instant`'s are guaranteed to be at millisecond granularity across systems? I'm not sure that subtracting `timestamp_frequency` will always do what we want. I'm reading through docs now